pkg/blob/nodeserver.go

```diff
@@ -278,6 +303,17 @@ func (d *Driver) NodeStageVolume(ctx context.Context, req *csi.NodeStageVolumeRe
 		serverAddress = fmt.Sprintf("%s.blob.%s", accountName, storageEndpointSuffix)
 	}
 
+	if isCacheEnabled {
+		klog.V(2).Infof("cache enabled, edge cache will be used")
+		if err := d.edgeCacheManager.EnsureVolume(accountName, accountKey, containerName, targetPath); err != nil {
```

Comment:
this is also creating a volume in NodeStageVolume, it's not appropiate to create volume here since driver running on the agent node may probably do not have permission to create volume.

Reply:
To clarify, this is creating an edge cache volume, not a kubernetes volume resource. We have to request the edge cache create a mount point for this blob storage resource.

I believe we need this in NodeStageVolume for static provisioning cases. When a statically provisioned PV is requested by the kube op, the first time the CSI driver will be notified is during nodestage volume. When caching is requested, we need to notify the edge cache services to create the mount point, then follow the alternate GRPC-based mount path.

I'm going to send along our design doc as well which should clarify some of this.

Reply:
Is it okay to resolve this? Hoping to get this PR merged and iterate further on our staging branch.

Reply:
I am ok with this, while if you merge the PR to a staging branch, are you going to make upstream changes finally? Maybe it's better merge this to upstream first if edge cache service is already ready?

Reply:
Currently edge cache is still in heavy development so we aren't ready to upstream yet.

Reply:
sure, let me know when you are ready.